### DISCUSSION TITLE
Tags as match objects. for the HTML and JSX peeps.

### DISCUSSION THREAD

**Discussion by KyleFontenot** [helix-editor/helix#1870] [Ideas]
`mit` would be nice to grab everything inside the HTML **t**ag. Not seeing any conflicts in match mode, but not sure how reliant on treesitter something like this is, though **tags** in XML, HTML, JSX are specifically with `<>` surrounds so it seems reasonable without needing a grammar.

**Comment by the-mikedavis** [comment]
You can do this currently with `mi<` or `mi>` to select within the angle brackets of `<tags>`

(though it does select the `/` in a close `</tag>` which might not be ideal depending on what you want to select)

**Reply by bcspragu** [reply]
Not sure what @KyleFontenot's initial intention was, but I think the common usecase for something like:

```
<tag>The Contents</tag>
```

Is being able to select `The Contents`. In Vim/NeoVim with something like [vim-surround](https://github.com/tpope/vim-surround), one can do `sit` (surround inside tag) to select `The Contents`. In Helix, it'd be something like `mit`, as @KyleFontenot mentioned above.

A workaround that I've been using is `A-i` while the cursor is in the tag body, which will select all of the contents.

**Comment by ff2400t** [comment]
I have a PR for this #3282 , which uses tree-sitter queries where we use `e` for for selection instead of `t`

**Comment by emirom** [comment]
when your cursor is between tags, the `Alt` and `i` keys select the contents of the tags.
